_posts/en/newsletters/2021-07-14-newsletter.md

```diff
@@ -0,0 +1,208 @@
+  a4="It uses the witness script instead of its hash in the P2SH-P2WSH redeem script."
+  a4link="https://bitcoincore.reviews/22363#153-158"
+%}
+
+## Preparing for taproot #4: from P2WPKH to single-sig P2TR
+
+*A weekly [series][series preparing for taproot] about how developers
+and service providers can prepare for the upcoming activation of taproot
+at block height {{site.trb}}.*
+
+{% include specials/taproot/en/03-p2wpkh-to-p2tr.md %}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [LND 0.13.1-beta.rc1][LND 0.13.1-beta] is a maintenance release with
+  minor improvements and bug fixes for features introduced in
+  0.13.0-beta.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [C-Lightning #4625][] updates its LN offers implementation to match the latest
+  [specification changes][offers spec changes]. Notably, offers are no longer
+  required to contain a signature. This significantly shortens the encoded
+  string for offers, improving QR code recognizability.
+
+- [Eclair #1746][] postgres support FIXME:Xekyo
+
+- [LND #5447][] adds a [document][lnd leader] describing how to set up
+  multiple LND nodes in a cluster with an alternative database that is
+  replicated between the cluster's nodes and which allows for automatic
+  failover.  Interested readers may wish to contrast this with the
+  approach taken by Eclair and described in [Newsletter #128][news128
+  eclair akka].
+
+- [Libsecp256k1 #844][] makes several updates to the API for [schnorr
+  signatures][topic schnorr signatures].  Most notable is a
+  [commit][nick varsig] that allows signing and verifying messages of
+  any length.  All current uses of signatures in Bitcoin sign a 32-byte
+  hash, but allowing signing of variable-length data could be useful for
+  applications outside of Bitcoin or to enable a new opcode such as
+  [OP_CHECKSIGFROMSTACK][topic op_checksigfromstack] to [verify
+  signatures][oconnor var csfs] created for non-Bitcoin systems.  It's
+  expected that the [BIP340][] specification of schnorr signatures for
+  Bitcoin will be updated to describe safely signing variable length
+  data.
+
+- [BIPs #943][] ajtowns/bip-anyprevout FIXME:adamjonas
+
+- [BTCPay Server #2655][] signals to web browsers that they should not
+  send the HTTP `referer` <!-- sic --> field when the user clicks on an
+  link to a transaction explorer.  This avoids telling the block
```

Comment:
Suggest adding link to block explorer topic.

Reply:
fixed